### ISSUE TITLE
Why 1D diffusion is so extremely slow??

### ISSUE THREAD

**Issue by sagagk** [lucidrains/denoising-diffusion-pytorch#316]
In the demo 2D:

 46/1000 [03:07<1:03:57,  4.02s/it]

1 hour..

in a simple 1D example (1 channel and sequence of 128):

loss: 0.2947:   0%
 178/700000 [01:07<84:55:21,  2.29it/s]

84 hours!.

--------------------

Code:

import torch
from denoising_diffusion_pytorch import Unet1D, GaussianDiffusion1D, Trainer1D, Dataset1D

model = Unet1D(
    dim = 64,
    dim_mults = (1, 2, 4, 8),
    channels = 1
)

diffusion = GaussianDiffusion1D(
    model,
    seq_length = 128,
    timesteps = 1000,
    objective = 'pred_v'
)

training_seq = torch.rand(64, 1, 128) # features are normalized from 0 to 1

loss = diffusion(training_seq)
loss.backward()

# Or using trainer

dataset = Dataset1D(training_seq)  # this is just an example, but you can formulate your own Dataset and pass it into the `Trainer1D` below

trainer = Trainer1D(
    diffusion,
    dataset = dataset,
    train_batch_size = 32,
    train_lr = 8e-5,
    train_num_steps = 700000,         # total training steps
    gradient_accumulate_every = 2,    # gradient accumulation steps
    ema_decay = 0.995,                # exponential moving average decay
    amp = True,                       # turn on mixed precision
)
trainer.train()

# after a lot of training

sampled_seq = diffusion.sample(batch_size = 4)
sampled_seq.shape # (4, 32, 128)

**Comment by sagagk** [comment]
Oh sorry. It is the number of steps. :)

**Comment by LRQ577** [comment]
Have you solved the speed problem?
